flake.nix

```diff
@@ -18,7 +18,11 @@
           libgit2
           libz
         ];
-
+        lib = nixpkgs.lib;
+        rustToolchain = fenix.packages.${system}.fromToolchainName {
+          name = (lib.importTOML ./rust-toolchain.toml).toolchain.channel;
+          sha256 = "sha256-f/CVA1EC61EWbh0SjaRNhLL0Ypx2ObupbzigZp8NmL4=";
```

Comment:
Does it mean that we need to update the sha256 if we want to bump the toolchain?

Reply:
Yes, it's designed to ensure all dependencies are tracked by nix. If toolchain file changed, there will be a checksum mismatch.